targets/TARGET_Cypress/TARGET_PSOC6/cy_crc_api.c

```diff
@@ -31,6 +31,14 @@ static cyhal_crc_t cy_crc;
 static crc_algorithm_t cy_crc_cfg;
 static bool cy_crc_initialized = false;
 
+// Reverses width least significant bits of 32 bit input. Any bits more
+// significant than width are dropped.
+static uint32_t reverse(uint8_t width, uint32_t in)
+{
+    MBED_ASSERT(width <= 32);
```

Comment:
Personally, I wouldn't bother with the assert - this is internal, and you've already cleared the `HAL_CRC_IS_SUPPORTED` check, and there are a squillion `uint32_t`s around here. 32-bit or less is kind of assumed.

Actually though, having this after `HAL_CRC_IS_SUPPORTED` means the compiler should hopefully skip putting in the assert anyway. So maybe I shouldn't be worrying about code size.